--- sharutils-4.2.orig/src/uudecode.c
+++ sharutils-4.2/src/uudecode.c
@@ -292,12 +292,12 @@
 
       if (strncmp (buf, "begin", 5) == 0)
 	{
-	  if (sscanf (buf, "begin-base64 %o %s", &mode, buf) == 2)
+	  if (sscanf (buf, "begin-base64 %o %[^\n]", &mode, buf) == 2)
 	    {
 	      do_base64 = 1;
 	      break;
 	    }
-	  else if (sscanf (buf, "begin %o %s", &mode, buf) == 2)
+	  else if (sscanf (buf, "begin %o %[^\n]", &mode, buf) == 2)
 	    break;
 	}
     }
