--- sharutils-4.14/src/shar.c.orig	2013-10-13 00:18:28.000000000 +0200
+++ sharutils-4.14/src/shar.c	2013-10-20 08:11:36.027842920 +0200
@@ -926,7 +926,7 @@
         free (c_dir);
       }
     else
-      error (0, errno, _("Cannot get current directory name"));
+      error (0, errno, "%s", _("Cannot get current directory name"));
   }
 }
 
@@ -2148,7 +2148,7 @@
        */
       FILE * fp = freopen ("/dev/null", fwriteonly_mode, stderr);
       if (fp != stderr)
-        error (SHAR_EXIT_FAILED, errno,
+        error (SHAR_EXIT_FAILED, errno, "%s", 
                _("reopening stderr to /dev/null"));
     }
 
@@ -2219,7 +2219,7 @@
   if (HAVE_OPT(QUERY_USER))
     {
       if (HAVE_OPT(NET_HEADERS))
-	error (0, 0, _("PLEASE avoid -X shars on Usenet or public networks"));
+	error (0, 0, "%s", _("PLEASE avoid -X shars on Usenet or public networks"));
 
       fputs ("shar_wish=\n", output);
     }
