--- sharutils-4.7/m4/xstrtoimax.m4.orig	2005-04-26 04:08:50.000000000 +0200
+++ sharutils-4.7/m4/xstrtoimax.m4	2007-12-12 18:07:28.601169266 +0100
@@ -7,7 +7,7 @@
 [
   AC_REQUIRE([gt_TYPE_INTMAX_T])
   AC_REQUIRE([gl_AC_TYPE_UINTMAX_T])
-  AC_REQUIRE([gl_AC_TYPE_LONG_LONG])
+  AC_REQUIRE([AC_TYPE_LONG_LONG_INT])
   AC_REQUIRE([gl_AC_TYPE_UNSIGNED_LONG_LONG])
   AC_CHECK_DECLS([strtol, strtoul, strtoll, strtoimax, strtoumax])
   AC_CHECK_HEADERS(limits.h stdlib.h inttypes.h)
