diff -urNp -x '*.orig' sharutils-4.15.2.org/src/shar.c sharutils-4.15.2/src/shar.c
--- sharutils-4.15.2.org/src/shar.c	2015-05-22 17:27:34.000000000 +0200
+++ sharutils-4.15.2/src/shar.c	2024-12-04 00:00:06.485290021 +0100
@@ -931,7 +931,7 @@ print_header_stamp (FILE * fp)
         free (c_dir);
       }
     else
-      error (0, errno, _("Cannot get current directory name"));
+      error (0, errno, "%s", _("Cannot get current directory name"));
   }
 }
 
@@ -2154,7 +2154,7 @@ configure_shar (int * argc_p, char *** a
        */
       FILE * fp = freopen ("/dev/null", "w" FOPEN_BINARY, stderr);
       if (fp != stderr)
-        error (SHAR_EXIT_FAILED, errno,
+        error (SHAR_EXIT_FAILED, errno, "%s", 
                _("reopening stderr to /dev/null"));
     }
 
@@ -2225,7 +2225,7 @@ configure_shar (int * argc_p, char *** a
   if (HAVE_OPT(QUERY_USER))
     {
       if (HAVE_OPT(NET_HEADERS))
-	error (0, 0, _("PLEASE avoid -X shars on Usenet or public networks"));
+	error (0, 0, "%s", _("PLEASE avoid -X shars on Usenet or public networks"));
 
       fputs ("shar_wish=\n", output);
     }
