--- sharutils-4.2.orig/src/mailshar.in
+++ sharutils-4.2/src/mailshar.in
@@ -33,7 +33,7 @@
 If none of -MTBzZ are given, -z is automatically selected if *none*
 of the FILEs have an .arc, .exz, .gif, .z, .gz, .Z, .zip or .zoo suffix."
 
-temp=/usr/tmp/$$.shar
+temp=`tempfile`
 
 ### Decode the options.
 
--- sharutils-4.2.orig/src/remsync.in
+++ sharutils-4.2/src/remsync.in
@@ -1657,7 +1657,10 @@
 	}
     }
 
-    open (SCAN, ("find$list -type f -print 2> /tmp/$$.find"
+    $findtempfile = `tempfile`;
+    chop $findtempfile;
+
+    open (SCAN, ("find$list -type f -print 2> $findtempfile"
 		 . " | xargs $sum_command |"))
 	|| &interrupt ("Cannot launch program \`find\'");
 
@@ -1689,7 +1692,7 @@
 
     # Clean out scanning for inexisting files.
 
-    open (SCAN, "/tmp/$$.find");
+    open (SCAN, "$findtempfile");
     while (<SCAN>)
     {
 	chop;
@@ -1706,7 +1709,7 @@
 		&query ("Should I delete this scan (y/n)? [y]");
 		if (/^(y|yes)$/i)
 		{
-		    &delete_scan ($file);
+		    &command_delete_scan ($file);
 		}
 		else
 		{
@@ -1720,7 +1723,7 @@
 	}
     }
     close SCAN;
-    unlink "/tmp/$$.find";
+    unlink "$findtempfile";
 
     $study_files = 0;
 }
