--- v1/src/shar.c	Wed Apr  7 00:06:23 2004
+++ v2/src/shar.c	Wed Apr  7 00:06:49 2004
@@ -1905,7 +1905,12 @@
 	break;
 
       case 'o':
-	strcpy (output_base_name, optarg);
+	strncpy (output_base_name, optarg,sizeof(output_base_name));
+	if(strlen(optarg) >= sizeof(output_base_name))
+	{
+		printf("Error! Output prefix too long.\n");
+		exit(1);
+	}
 	if (!strchr (output_base_name, '%'))
 	  strcat (output_base_name, ".%02d");
 	part_number = 0;
