diff -Nru sharutils-4.2.1/configure.in sharutils-4.2.1.new/configure.in
--- sharutils-4.2.1/configure.in	Wed Oct 25 05:31:02 2000
+++ sharutils-4.2.1.new/configure.in	Wed Oct 25 05:31:45 2000
@@ -34,7 +34,7 @@
 AC_SUBST(DIST_ALPHA)
 
 AC_PROG_CC
-fp_PROG_INSTALL
+AC_PROG_INSTALL
 AC_PROG_MAKE_SET
 AC_PROG_RANLIB
 
@@ -64,11 +64,10 @@
 AC_AIX
 AC_ISC_POSIX
 AC_MINIX
-AC_C_CROSS
 if test $cross_compiling = no; then
   AC_C_CHAR_UNSIGNED
 fi
-fp_C_PROTOTYPES
+AM_C_PROTOTYPES
 AC_C_CONST
 AC_C_INLINE
 AC_C_BIGENDIAN
@@ -115,8 +114,8 @@
   MAN5PAGES="$MAN5PAGES uuencode"
 ])
 
-ud_GNU_GETTEXT
-fp_WITH_DMALLOC
+AM_GNU_GETTEXT
+AM_WITH_DMALLOC
 
 # sharutils is somewhat special about internationalization in that it
 # always requires the GNU .mo files installed.
@@ -129,10 +128,6 @@
   CATALOGS="$new_CATALOGS"
 fi
 
-AC_LINK_FILES($nls_cv_header_libgt, $nls_cv_header_intl)
-
-AC_OUTPUT([Makefile src/mail-files src/mailshar src/remsync contrib/Makefile \
-	   lib/Makefile intl/Makefile intl/po2tbl.sed src/Makefile \
-	   po/Makefile.in doc/Makefile checks/Makefile],
-[sed -e "/POTFILES =/r po/POTFILES" po/Makefile.in > po/Makefile; \
- test -z "$CONFIG_HEADERS" || echo timestamp > stamp-h])
+AC_OUTPUT(Makefile src/mail-files src/mailshar src/remsync contrib/Makefile \
+	lib/Makefile intl/Makefile intl/po2tbl.sed src/Makefile \
+	po/Makefile.in doc/Makefile checks/Makefile)
diff -Nru sharutils-4.2.1/Makefile.in sharutils-4.2.1.new/Makefile.in
--- sharutils-4.2.1/Makefile.in	Wed Oct 25 05:35:28 2000
+++ sharutils-4.2.1.new/Makefile.in	Wed Oct 25 05:36:01 2000
@@ -29,7 +29,7 @@
 acconfig.h aclocal.m4 config.h.in configure configure.in install-sh \
 mkinstalldirs stamp-h.in @DIST_ALPHA@
 
-SUBDIRS = doc lib @POSUB@ @INTLSUB@ src checks contrib
+SUBDIRS = doc lib @POSUB@ intl src checks contrib
 
 all: all-recursive
 
