diff -Nru sharutils-4.2/Makefile.in sharutils-4.2.new/Makefile.in
--- sharutils-4.2/Makefile.in	Sat Dec  2 03:34:42 1995
+++ sharutils-4.2.new/Makefile.in	Mon Jun 21 16:27:39 1999
@@ -35,7 +35,7 @@
 
 all-recursive install install-exec install-data uninstall TAGS ID \
 mostlyclean-recursive clean-recursive distclean-recursive \
-maintainer-clean-recursive check: config.h Makefile
+maintainer-clean-recursive check:
 	for subdir in $(SUBDIRS); do \
 	  target=`echo $@|sed 's/-recursive//'`; \
 	  echo making $$target in $$subdir; \
@@ -91,10 +91,6 @@
 dvi info:
 	cd doc && $(MAKE) $@
 
-# For understanding the following see the autoconf manual.
-configure: configure.in aclocal.m4
-	cd $(srcdir) && autoconf
-
 # autoheader might not change config.h.in
 config.h.in: stamp-h.in
 stamp-h.in: configure.in acconfig.h
@@ -104,12 +104,6 @@
 	test ! -f stamp-h || ./config.status
 	rm -f stamp-h && echo timestamp > stamp-h
 
-Makefile: Makefile.in config.status
-	./config.status
-
-config.status: configure config.h.in
-	./config.status --recheck
-
 # Tell versions [3.59,3.63) of GNU make not to export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
--- sharutils-4.2/Makefile.in	Mon Jun 21 16:50:49 1999
+++ sharutils-4.2.new/Makefile.in	Mon Jun 21 16:51:48 1999
@@ -91,12 +91,6 @@
 dvi info:
 	cd doc && $(MAKE) $@
 
-# autoheader might not change config.h.in
-config.h.in: stamp-h.in
-stamp-h.in: configure.in acconfig.h
-	cd $(srcdir) && autoheader
-	rm -f $(srcdir)/stamp-h.in && echo timestamp > $(srcdir)/stamp-h.in
-
 #config.status might not change config.h
 # Don't rerun config.status if we just configured.
 config.h: stamp-h
